modules/frontend/metrics_query_range_sharder.go

```diff
@@ -38,10 +39,15 @@ type QueryRangeSharderConfig struct {
 	MaxDuration           time.Duration `yaml:"max_duration"`
 	QueryBackendAfter     time.Duration `yaml:"query_backend_after,omitempty"`
 	Interval              time.Duration `yaml:"interval,omitempty"`
+	QueryGeneratorBlocks  bool          `yaml:"query_generator_blocks,omitempty"`
```

Comment:
Would probably rename this to `rf1_read_path`, and then the logic related to setting things like `replicationFactor=1` is also clearer.